sim/dex-data.ts

```diff
@@ -34,7 +34,21 @@ export function toID(text: any): ID {
 	/* eslint-enable @typescript-eslint/prefer-optional-chain */
 }
 
-export class BasicEffect implements EffectData {
+/**
+ * Like Object.assign but only assigns fields missing from self.
+ * Facilitates consistent field ordering in constructors.
+ * Modifies self in-place.
+ */
+export function assignNewFields(self: AnyObject, data: AnyObject) {
```

Comment:
`assignMissingFields` would be clearer to me, I didn't understand what this did until I read the code.